test/testnullpointer.cpp

```diff
@@ -1642,6 +1643,26 @@ class TestNullPointer : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void nullpointer49() {
+        check("int f();\n"
+              "bool g();\n"
+              "void h(A *  a) {\n"
```

Comment:
This testcase is a bit long and hard to read. I didn't immediately see what is tested. Well maybe it can be simplified a bit. Maybe a value `10` instead of a function call `f()` can be used. Maybe `g()` can be replaced by a argument (saves 1 line). Maybe the flag `h` could get a better name such as `found`.

In my head, it's not the null pointer check that is tested here but primarily how `ValueFlow` tracks null pointers. Therefore I do not think of this as a "unit test". I know there was discussions before to separate such tests somehow. With a comment or naming convention.. I don't remember.

I feel that the FP has nothing to do with the null pointer check. You'd get such FP for any check that uses `ValueFlow` if `ValueFlow` does not work. We could just as well write such tests for division by zero and lots of other checks instead... Like:

        check("int f();\n"
              "bool g();\n"
              "void h(int *  a) {\n"
              "    bool h = false;\n"
              "    int x = 0;\n"
              "    for (int i = 0; i < f(); i++) {\n"
              "        if (g()) {\n"
              "            h = true;\n"
              "            x = a[ i ];\n"
              "            break;\n"
              "        }\n"
              "    }\n"
              "    if (h) {\n"
              "        A e(foo / x);\n"   // <- division by zero?
              "    }\n"
              "}\n");
        ASSERT_EQUALS("", errout.str());

.. I am not against that we add your test. I just have a feeling this is not a good structure.

Reply:
I am hoping that we can change the structure somewhat someday. So test cases are grouped logically. For now.. if you add some comment that is fine for me. Not sure is `// system test` a good comment for this? feel free to give another suggestion.

Reply:
or the testcase could be named `valueFlow..`.

Reply:
@pfultz2  Any chance to get the conflicts resolved and add a comment/rename the testcase?